crates/biome_js_analyze/src/lint/nursery/use_for_component.rs

```diff
@@ -0,0 +1,114 @@
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// let Component = (props) => <ol>{props.data.map(d => <li>{d.text}</li>)}</ol>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// let Component = (props) => <>{props.data.map(d => <li>{d.text}</li>)}</>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// let Component = (props) => (
+    ///   <ol>
+    ///     {props.data.map((d) => (
+    ///       <li key={d.id}>{d.text}</li>
+    ///     ))}
+    ///   </ol>
+    /// );
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// let Component = (props) => <ol><For each={props.data}>{d => <li>{d.text}</li>}</For></ol>;
+    /// ```
+    ///
+    /// ```jsx
+    /// let abc = x.map(y => y + z);
+    /// ```
+    ///
+    /// ```jsx
+    /// let Component = (props) => {
+    ///  let abc = x.map(y => y + z);
+    ///  return <div>Hello, world!</div>;
+    /// }
+    /// ```
+    ///
+    pub UseForComponent {
+        version: "next",
+        name: "useForComponent",
+        language: "js",
+        domains: &[RuleDomain::Solid],
+        recommended: false,
+        sources: &[RuleSource::EslintSolid("perfer-for")],
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+impl Rule for UseForComponent {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if let Some(parent) = node.parent::<JsxExpressionChild>() {
+            // Only judge the expression child under JSX_CHILD_LIST
+            // all jsxexpression with case can be covered here like:
+            // <ol>{props.data.map(d => <li>{d.text}</li>)}</ol>
+            if !matches!(
+                parent.syntax().parent().kind(),
+                Some(JsSyntaxKind::JSX_CHILD_LIST)
+            ) {
+                return None;
+            }
+
+            // check for Array.prototype.map in JSX
+            let member_expression =
+                AnyJsMemberExpression::cast(node.callee().ok()?.omit_parentheses().into_syntax())?;
+            let args = node.arguments().ok()?.args();
+
+            if args.len() != 1 || member_expression.member_name()?.text() != "map" {
+                return None;
+            }
+
+            return Some(());
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.syntax().text_trimmed_range(),
+            markup! {
+            "Array.prototype.map will cause DOM elements to be recreated, it is not recommended to use it in Solid here."
+            },
+        ).note(markup! {
+            "Use Solid's [`<For />`](https://docs.solidjs.com/reference/components/for) component for efficiently rendering lists."
```

Comment:
Probably you aren't aware, but our markup doesn't use markdown. However, we have `Hyperlink` tag already, here's an example: 

https://github.com/biomejs/biome/blob/19238f1e004ca4987d2a64a8f5c92f3645b5ded2/crates/biome_js_analyze/src/lint/a11y/no_header_scope.rs#L84-L85

Reply:
done